src/GraphMsg.chpl

```diff
@@ -0,0 +1,4510 @@
+                    }
+
+
+                    //SetCurF<=>SetNextF;
+                    SetCurF=SetNextF;
+                    SetNextF.clear();
+              }//end while  
+              if (currentindex+1<Nv) {
+                 forall i in 0..Nv-1 with(+reduce currentindex) {
+                     if depth[i]==-1 {
+                       cmary[currentindex+1]=i;
+                       currentindex+=1;  
+                     }
+                 }
+              }
+              cmary.reverse();
+              forall i in 0..Nv-1{
+                  indexary[cmary[i]]=i;
+              }
+    
+              var tmpary:[0..Ne-1] int;
+              forall i in 0..Ne-1 {
+                      tmpary[i]=indexary[src[i]];
+              }
+              src=tmpary;
+              forall i in 0..Ne-1 {
+                      tmpary[i]=indexary[dst[i]];
+              }
+              dst=tmpary;
+ 
+              return "success";
+          }//end RCM_u
+
+
+          coforall loc in Locales  {
+              on loc {
+                  forall i in srcR.localSubdomain(){
+                        srcR[i]=dst[i];
+                        dstR[i]=src[i];
+                   }
+              }
+          }
+          combine_sortR();
+          set_neighbourR();
+
+          if (DegreeSortFlag>0) {
+             var DegreeArray=makeDistArray(Nv,int);
+             var VertexArray=makeDistArray(Nv,int);
+             var tmpedge=makeDistArray(Ne,int);
+             coforall loc in Locales  {
+                on loc {
+                  forall i in neighbour.localSubdomain(){
+                        DegreeArray[i]=neighbour[i]+neighbourR[i];
+                        //writeln("Degree of vertex ",i," =",DegreeArray[i]," =",neighbour[i]," +",neighbourR[i]);
+                   }
+                }
+             }
+             //writeln("degree array=",DegreeArray);
+             var tmpiv = argsortDefault(DegreeArray);
+             forall i in 0..Nv-1 {
+                 VertexArray[tmpiv[i]]=i;
+                 //writeln("Old vertex",tmpiv[i], " -> ",i);
+             }
+             //writeln("relabeled vertex array=",VertexArray);
+             coforall loc in Locales  {
+                on loc {
+                  forall i in src.localSubdomain(){
+                        tmpedge[i]=VertexArray[src[i]];
+                  }
+                }
+             }
+             src=tmpedge;
+             coforall loc in Locales  {
+                on loc {
+                  forall i in dst.localSubdomain(){
+                        tmpedge[i]=VertexArray[dst[i]];
+                  }
+                }
+             }
+             dst=tmpedge;
+             coforall loc in Locales  {
+                on loc {
+                  forall i in src.localSubdomain(){
+                        if src[i]>dst[i] {
+                           var tmpx=src[i];
+                           src[i]=dst[i];
+                           dst[i]=tmpx;
```

Comment:
Optimization:  You could use the swap operator here `src[i] <=> dst[i]` should be equivalent to using a tmp variable to store the value to swap.

Reply:
Thanks and I did not know the <=> operator can be used here.